packages/core-sdk/resolvers/token-resolvers/resolvers/uniswap/src/query/functions/getTokenComponents.ts

```diff
@@ -13,6 +13,28 @@ import {
   Token_TokenType,
 } from "../w3";
 
+function getPairTokenAddresses(pairAddress: string, connection: Ethereum_Connection): string[] {
+  // get token addresses
+  const token0AddressResult = Ethereum_Query.callContractView({
+    address: pairAddress,
+    method: "function token0() external view returns (address)",
+    args: [],
+    connection: connection,
+  });
+  // if exception encountered, pair contract presumed not to exist
+  if (token0AddressResult.isErr) {
+    throw new Error("Invalid protocol token");
+  }
+  const token0Address = token0AddressResult.unwrap();
+  const token1Address = Ethereum_Query.callContractView({
+    address: pairAddress,
+    method: "function token1() external view returns (address)",
+    args: [],
+    connection: connection,
+  }).unwrap();
```

Comment:
It is not possible in this case, because the pair address is deterministic. If the pool contract exists, it will have `token0` and `token1` methods. If it does not exist, it will have neither method.